### ISSUE TITLE
Add a JobVersion API to expose version information

### ISSUE THREAD

**Issue by collado-mike** [MarquezProject/marquez#1455]
Proposal for a couple of new APIs based on work already started by @wslulciuc 

Some open questions- 
* should we bother including the `context` field in the job version here? Seems weird to add a new type with an already deprecated field.
* should we include a`latestRun` field, as in the job API? Seems like it would be a useful field to include.

```yaml

paths:
  /namespaces/{namespace}/jobs/{job}/versions:
    parameters:
      - $ref: '#/components/parameters/namespace'
      - $ref: '#/components/parameters/job'
    get:
      summary: get a list of versions for a given job name
      description: returns a list of versions for a given job name and namespace
      tags:
        - Jobs
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/JOB_VERSION_LIST'
  /namespaces/{namespace}/jobs/{job}/versions/{version}:
    parameters:
      - $ref: '#/components/parameters/namespace'
      - $ref: '#/components/parameters/job'
      - $ref: '#/components/parameters/version'
    get:
      summary: get the details of a given job name
      description: returns a specified version for a given job name and namespace
      tags:
        - Jobs
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/JOB_VERSION'

components:
  schemas:
    JOB_VERSION_LIST:
      type: object
      properties:
        versions:
          items:
            $ref: '#/components/schemas/JOB_VERSION'

    JOB_VERSION:
      type: object
      properties:
        id:
          description: The ID of the job.
          type: object
          properties:
            namespace:
              type: string
              description: The namespace of the job.
            name:
              type: string
              description: The name of the job.
            version:
              type: string
              description: The version of the job
        name:
          description: The name of the job.
          type: string
        version:
          description: The version of the job
          type: string
        createdAt:
          description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the job version was created.
          type: string
          format: date-time
        namespace:
          description: The namespace of the job.
          type: string
        inputs:
          description: The set of **input** dataset ids.
          type: array
          items:
            type: string
            description: the dataset UUID for each input dataset
            format: UUID
          uniqueItems: true
        outputs:
          description: The set of **output** dataset ids.
          type: array
          items:
            type: string
            description: the dataset UUID for each output dataset
            format: UUID
          uniqueItems: true
        location:
          description: The URL of the job source code or artifact for the given job version.
          type: string
          format: URL
        context:
          description: Deprecated- Use facet. A key/value pair that must be of type `string`. A context can be used for getting additional details about the job.
          type: object
          deprecated: true
        facets:
          $ref: '#/components/schemas/jobFacets'
      example:
        id:
          namespace: my-namespace
          name: my-job
          version: 56472c57-a2ef-4218-b7b7-d2af02a343fd
        type: BATCH
        name: my-job
        version: 56472c57-a2ef-4218-b7b7-d2af02a343fd
        createdAt: 2019-05-09T19:49:24.201361Z
        namespace: my-namespace
        inputs: ["7b10f8dd-6091-403d-ab86-20de85d23a4a", "95eaa980-0894-4f66-bccb-64a6697407ad"]
        outputs: ["01d658b0-d1ab-4842-89bb-3ba2edb9f65d"]
        location: https://github.com/my-jobs/blob/124f6089ad4c5fcbb1d7b33cbb5d3a9521c5d32c
        context: {'SQL': "SELECT * FROM mytable;"}
        description: My first job!
        facets: {
          'context': {
            '_producer': 'https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client',
            '_schemaURL': 'https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/spec/OpenLineage.yml#MyCustomJobFacet'
          }
        }
```

**Comment by wslulciuc** [comment]
> should we bother including the context field in the job version here? Seems weird to add a new type with an already deprecated field.

I would keep the job context around until it's official removed in `0.17.0`. That way the behavior matches the current API model. 

> should we include a latestRun field, as in the job API? Seems like it would be a useful field to include.

I would make sure to document that `latestRun` is the latest run associated with the job version given that a job version can have one or more runs. This is different that the `latestRun` for a Job.

**Comment by collado-mike** [comment]
Updated types based on feedback. Also, I wanted to be consistent wrt the input/output datasets as returned in other job APIs, so I changed to spec to return a `datasetId`, defined below
```diff
+   datasetId:
+     type: object
+     description: The ID of the dataset.
+     properties:
+       namespace:
+         type: string
+         description: The namespace of the dataset.
+       name:
+         type: string
+         description: The name of the dataset.

    jobVersionList:
      type: object
      properties:
        versions:
          items:
            $ref: '#/components/schemas/JOB_VERSION'

    jobVersion:
      type: object
      properties:
        id:
          description: The ID of the job.
          type: object
          properties:
            namespace:
              type: string
              description: The namespace of the job.
            name:
              type: string
              description: The name of the job.
            version:
              type: string
              description: The version of the job
        name:
          description: The name of the job.
          type: string
        version:
          description: The version of the job
          type: string
        createdAt:
          description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the job version was created.
          type: string
          format: date-time
        namespace:
          description: The namespace of the job.
          type: string
        inputs:
-        description: The set of **input** dataset ids.
+        description: The set of **input** datasets
          type: array
          items:
-          type: string
-          description: the dataset UUID for each input dataset
-          format: UUID
+          $ref: '#/components/schemas/datasetId'
          uniqueItems: true
        outputs:
-        description: The set of **output** dataset ids.
+        description: The set of **output** datasets
          type: array
          items:
-          type: string
-          description: the dataset UUID for each output dataset
-          format: UUID
+          $ref: '#/components/schemas/datasetId'
          uniqueItems: true
        location:
          description: The URL of the job source code or artifact for the given job version.
          type: string
          format: URL
        context:
          description: Deprecated- Use facet. A key/value pair that must be of type `string`. A context can be used for getting additional details about the job.
          type: object
          deprecated: true
+      latestRun:
+        description: The latest run associated with the job version.
+        type: object
+        properties:
+          id:
+            description: The ID of the run.
+            type: string
+            format: uuid
+          createdAt:
+            description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the run was created.
+            type: string
+            format: date-time
+          updatedAt:
+            description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the run was updated.
+            type: string
+            format: date-time
+          nominalStartTime:
+            description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal start time of the run.
+            type: string
+            format: date-time
+          nominalEndTime:
+            description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal end time of the run.
+            type: string
+            format: date-time
+          state:
+            description: The current state of the run.
+            type: string
+            enum: [ NEW, RUNNING, COMPLETED, FAILED, ABORTED ]
+          startedAt:
+            type: integer
+            description: The time at which the run was started. Measured in seconds since the Unix epoch.
+          endedAt:
+            type: integer
+            description: The time at which the run was ended. Measured in seconds since the Unix epoch.
+          durationMs:
+            type: integer
+            description: The total duration of the run. Measured in milliseconds.
+          args:
+            description: The arguments of the run.
+            type: object
        facets:
          $ref: '#/components/schemas/jobFacets'
      example:
        id:
          namespace: my-namespace
          name: my-job
          version: 56472c57-a2ef-4218-b7b7-d2af02a343fd
        type: BATCH
        name: my-job
        version: 56472c57-a2ef-4218-b7b7-d2af02a343fd
        createdAt: 2019-05-09T19:49:24.201361Z
        namespace: my-namespace
        inputs: [{namespace: my-namespace, name: my-dataset}]
        outputs: []
        location: https://github.com/my-jobs/blob/124f6089ad4c5fcbb1d7b33cbb5d3a9521c5d32c
        context: {'SQL': "SELECT * FROM mytable;"}
        description: My first job!
        facets: {
          'context': {
            '_producer': 'https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client',
            '_schemaURL': 'https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/spec/OpenLineage.yml#MyCustomJobFacet'
          }
        }
```

**Comment by wslulciuc** [comment]
Fixed #1462
